airflow/secrets/local_filesystem.py

```diff
@@ -0,0 +1,257 @@
+
+
+def _parser_env_file(content: str) -> Tuple[Dict[str, List[str]], List[FileSyntaxError]]:
+    """
+    Parse a file in the ``.env '' format.
+
+   .. code-block:: text
+        MY_CONN_ID=my-conn-type://my-login:my-pa%2Fssword@my-host:5432/my-schema?param1=val1&param2=val2
+
+    :param content:
+    :return: Tuple with mapping of key and list of values and list of syntax errors
+    """
+    secrets: Dict[str, List[str]] = defaultdict(list)
+    errors: List[FileSyntaxError] = []
+    for line_no, line in enumerate(content.splitlines(), 1):
+        if not line:
+            # Ignore empty line
+            continue
+
+        if COMMENT_PATTERN.match(line):
+            # Ignore comments:
+            continue
+
+        var_parts: List[str] = line.split("=", 2)
+        if len(var_parts) != 2:
+            errors.append(
+                FileSyntaxError(
+                    line_no=line_no,
+                    message='Invalid line format. The line should contain at least one equal sign ("=").',
+                )
+            )
+            continue
+
+        key, value = var_parts
+        if not key:
+            errors.append(FileSyntaxError(line_no=line_no, message="Invalid line format. Key is empty.",))
+        secrets[key].append(value)
+    return secrets, errors
+
+
+def _parse_json_file(content: str) -> Tuple[Dict[str, Any], List[FileSyntaxError]]:
+    """
+    Parrse a file in the JSON format.
+
+    :param content:
+    :return: Tuple with mapping of key and list of values and list of syntax errors
+    """
+    if not content:
+        return {}, [FileSyntaxError(line_no=1, message="The file is empty.")]
+    try:
+        secrets = json.loads(content)
+    except JSONDecodeError as e:
+        return {}, [FileSyntaxError(line_no=int(e.lineno), message=e.msg)]
+    if not isinstance(secrets, dict):
+        return {}, [FileSyntaxError(line_no=1, message="The file should contain the object.")]
+
+    return secrets, []
+
+
+def _parse_secret_file(file_content: str, file_path: str) -> Tuple[Dict[str, Any], List[FileSyntaxError]]:
+    """
+    Based on the file extension format, selects a parser and parse the file.
+
+    :param file_content:
+    :param file_path:
+    :return:
+    """
+    if file_path.lower().endswith(".env"):
+        secrets, parse_errors = _parser_env_file(file_content)
+    elif file_path.lower().endswith(".json"):
+        secrets, parse_errors = _parse_json_file(file_content)
+    else:
+        raise AirflowException("Unsupported file format. The file must have the extension .env or .json")
+    return secrets, parse_errors
+
+
+def _read_secret_file(file_path: str) -> Dict[str, str]:
+    if not os.path.exists(file_path):
+        raise AirflowException(
+            f"File {file_path} was not found. Check the configuration of your secret backend."
+        )
+
+    log.debug("Reading file: %s", file_path)
+    with open(file_path) as f:
+        file_content = f.read()
```

Comment:
I think we should pass the file path directly to `_parser_env_file` and `_parse_json_file` and let both the method handle opening of files. 

How about using https://pypi.org/project/environs/ for reading `.env` file? 

Also please rename `_parser_env_file` to `_parse_env_file`

Reply:
I need to check if this library allows duplicate keys.

Reply:
environs use the .dotenv library to parse the file.  Unfortunately, this library does not support duplicate keys. In our case, it is necessary to handle multiple connections with the same connection ID.
https://github.com/theskumar/python-dotenv/blob/master/src/dotenv/main.py#L81